src/random-seed/random-seed.c

```diff
@@ -311,6 +315,101 @@ static int save_seed_file(
         return 0;
 }
 
+static int refresh_boot_seed(void) {
+        uint8_t buffer[RANDOM_EFI_SEED_SIZE];
+        struct sha256_ctx hash_state;
+        _cleanup_free_ void *seed_file_bytes = NULL;
+        _cleanup_free_ char *esp_path = NULL;
+        _cleanup_close_ int seed_fd = -1;
+        size_t len;
+        ssize_t r;
```

Comment:
hmmpf. all across our codebase we define `int r`. The variable has a special status for being the default choice for placing errno/return values into. Please do so here too. Let's not surprise people here...

Reply:
so I guess there should be two variables here:

```c
ssize_t n;
int r; 
```

and then for getrandom() and so on, use `n` and for the others `r`, depending on the type.

typesafety matters, and these two types will be different on most common archs after all

Reply:
Ack.